images/golang-dind/build.yaml

```diff
@@ -1,14 +1,10 @@
 name: golang-dind # Name of the image to be built
 
 variants:
-  "1.22":
+  "1.24":
```

Comment:
I don't think we have any location where we HAVE to use an older go version, just reduced the number of variants to just 1.24 here. The old images will remain in the registry btw.
The goal is to update all places where this image is used to 1.24 after merging this PR.

Reply:
Sounds good to me, we should adopt 1.24 everywhere soon anyway!